src/ProjectTemplates/BlazorWasm.ProjectTemplates/BlazorWasm-CSharp.Client.csproj.in

```diff
@@ -11,10 +11,25 @@
     <PackageReference Include="Microsoft.AspNetCore.Blazor.DevServer" Version="${MicrosoftAspNetCoreBlazorDevServerPackageVersion}" PrivateAssets="all" />
     <PackageReference Include="Microsoft.AspNetCore.Blazor.HttpClient" Version="${MicrosoftAspNetCoreBlazorHttpClientPackageVersion}" />
   </ItemGroup>
+
   <!--#if Hosted -->
   <ItemGroup>
     <ProjectReference Include="..\Shared\BlazorWasm-CSharp.Shared.csproj" />
   </ItemGroup>
+
   <!--#endif -->
+  <!--#if PWA -->
+  <Target Name="PublishServiceWorker" AfterTargets="CopyFilesToPublishDirectory">
+    <!-- Use the "published" version of the service worker in the output -->
+    <PropertyGroup>
+      <ServiceWorkerMoveFrom>$(PublishDir)$(BlazorPublishDistDir)service-worker.published.js</ServiceWorkerMoveFrom>
+      <ServiceWorkerMoveTo>$(PublishDir)$(BlazorPublishDistDir)service-worker.js</ServiceWorkerMoveTo>
```

Comment:
Is there a reason why this can't just be

```xml
 <ItemGroup>
   <Content Update="service-worker.js" CopyToOutputDirectory="false" CopyToPublishDirectory="false" />
   <ContentWithTargetPath Update="service-worker.published.js" CopyToOutputDirectory="false" CopyToPublishDirectory="true">
     <TargetPath>service-worker.js</TargetPath>
   <ContentWithTargetPath />
 </ItemGroup>
```

Reply:
And it doesn't have to be inside the target.

Reply:
👍  That looks better!

Reply:
Oh, except we do still need to have some way of making the file's contents get modified during publish so the browser knows to install a new service worker.